packages/ember-views/lib/components/component.js

```diff
@@ -456,6 +456,182 @@ var Component = View.extend(TargetActionSupport, {
     @public
     @property positionalParams
   */
+
+  /**
+    `didInitAttrs` runs on the first render after a component was
+    created and passed attrs are guaranteed to be present.
+
+    The attributes will be available as this.get('attrName') or by
+    accessing the `attrs` property of the options argument.
+
+    ```app/components/my-collection.js
+    export default Ember.Component.extend({
+      content: null,
+
+      didInitAttrs: function(options) {
+        var content = this.get('content');
+        if (!content) {
+          this.set('content', []);
+        }
+      }
+    });
+    ```
+
+    @public
+    @method didInitAttrs
+    @param options {Object}
+    @since 1.13.0
+  */
+
+  /**
+    `didReceiveAttrs` runs on the first render after `didInitAttrs`,
```

Comment:
Similarly to above, it is ran the first time from within the constructor before render and it is reran whenever the bound attributes change (not necessarily for every re-render).